File: src/main/java/me/arasple/mc/trhologram/listeners/ListenerPlayerJoin.java
package me.arasple.mc.trhologram.listeners;

import io.izzel.taboolib.module.inject.TListener;
import me.arasple.mc.trhologram.hologram.HologramManager;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerJoinEvent;

/**
 * @author Arasple
 * @date 2020/1/30 13:11
 */
@TListener
public class ListenerPlayerJoin implements Listener {

    @EventHandler
    public void onJoin(PlayerJoinEvent e) {
        Player player = e.getPlayer();
        HologramManager.getHolograms().stream().filter(hologram -> hologram.isVisible(player)).forEach(hologram -> hologram.display(player));
    }

}

File: src/main/java/me/arasple/mc/trhologram/listeners/ListenerPlayerJoin.kt
package me.arasple.mc.trhologram.listeners

import io.izzel.taboolib.module.inject.TListener
import me.arasple.mc.trhologram.hologram.Hologram
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerJoinEvent

/**
 * @author Arasple
 * @date 2020/2/14 9:13
 */
@TListener
class ListenerPlayerJoin : Listener {

    @EventHandler
    fun onJoin(e: PlayerJoinEvent) {
        Hologram.getHolograms().stream().filter { hologram -> hologram.isVisible(e.player) }.forEach { hologram -> hologram.display(e.player) }
    }

}
